app/Http/Controllers/PhotoController.php

```diff
@@ -93,6 +94,37 @@ public function add(AlbumIDRequest $request): Photo
 		/** @var UploadedFile $file */
 		$file = $request->file('0');
 		$sourceFileInfo = SourceFileInfo::createByUploadedFile($file);
+
+		// This code is a nasty work-around which should not exist.
+		// PHP stores a temporary copy of the uploaded file without a file
+		// extension.
+		// Unfortunately, most of our methods pass around absolute file paths
+		// instead of proper `File` object.
+		// During the process we have a lot of code which tries to
+		// re-determine the MIME type of the file based on the file path.
+		// This is not only inefficient, but the original MIME type (of the
+		// uploaded file) gets lost on the way.
+		// As a work-around we store the uploaded file with a file extension.
+		// Unfortunately, we cannot simply re-name the file, because this
+		// might break due to permission problems for certain installation
+		// if the temporarily uploaded file is stored in the system-global
+		// temporary directory below another mount point or another Docker
+		// image than the Lychee installation.
+		// Hence, we must make a deep copy.
```

Comment:
I feel like ideally we should try to rename and only if that fails, do a deep copy. But since this is a temporary fix, I'm OK with the way it is.